Sources/Edge.swift

```diff
@@ -94,37 +97,27 @@ public class Edge: NSObject, Extension {
             Log.warning(label: LOG_TAG, "handleExperienceEventRequest - An unexpected error has occurred, ECID is nil.")
         }
 
-        // get Assurance integration id and include it in to the requestHeaders
-        var requestHeaders: [String: String] = [:]
-        if let assuranceSharedState = getSharedState(extensionName: Constants.SharedState.Assurance.STATE_OWNER_NAME, event: event)?.value {
-            if let assuranceIntegrationId = assuranceSharedState[Constants.SharedState.Assurance.INTEGRATION_ID] as? String {
-                requestHeaders[Constants.NetworkKeys.HEADER_KEY_AEP_VALIDATION_TOKEN] = assuranceIntegrationId
-            }
-        }
-
         // Build and send the network request to Experience Edge
         let listOfEvents: [Event] = [event]
-        if let requestPayload = requestBuilder.getRequestPayload(listOfEvents) {
-            let requestId: String = UUID.init().uuidString
-
-            // NOTE: the order of these events needs to be maintained as they were sent in the network request
-            // otherwise the response callback cannot be matched
-            networkResponseHandler.addWaitingEvents(requestId: requestId,
-                                                    batchedEvents: listOfEvents)
-            guard let url: URL = networkService.buildUrl(requestType: ExperienceEdgeRequestType.interact,
-                                                         configId: configId,
-                                                         requestId: requestId) else {
-                Log.debug(label: LOG_TAG, "handleExperienceEventRequest - Failed to build the URL, dropping current event '\(event.id.uuidString)'.")
-                return
-            }
-
-            let callback: ResponseCallback = NetworkResponseCallback(requestId: requestId, responseHandler: networkResponseHandler)
-            networkService.doRequest(url: url,
-                                     requestBody: requestPayload,
-                                     requestHeaders: requestHeaders,
-                                     responseCallback: callback,
-                                     retryTimes: Constants.Defaults.NETWORK_REQUEST_MAX_RETRIES)
+        guard let requestPayload = requestBuilder.getRequestPayload(listOfEvents) else {
+            Log.debug(label: LOG_TAG,
+                      "handleExperienceEventRequest - Failed to build the request payload, dropping current event '\(event.id.uuidString)'.")
+            return
         }
+
+        let edgeHit = EdgeHit(configId: configId, requestId: UUID().uuidString, request: requestPayload, event: event)
```

Comment:
I've opted to build the request first as it is much safer than persisting the `Event`/list of events and decoding them later. The reason for this is that it could be possible that some of the event/xdm data is not compliant with `AnyCodable`, in the case that the event has data that can not be encoded/decoded with `AnyCodable`, that data is thrown out during encoding/decoding. I'd prefer to migrate the hits afterward then to hope that all the data is compliant with `AnyCodable` to minimize the risk of losing data.

For reference, the types that `AnyCodable` can handle can be found here:
https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/utility/AnyCodable.swift#L84

Reply:
Actually, this shouldn't make much of a difference as `EdgeRequest` contains `let events: [[String: AnyCodable]]?`. I'll update the PR.

Reply:
Do we think it's possible to have data that falls outside the bounds of AnyCodable?

Reply:
The allowed AnyCodable types should be sufficient, but there's always the chance a customer could pass in something else. I think we should document the allowed types in ExperienceEvent for cases where customers are building their own dictionaries or implementing their own Schema classes. I think we have a similar issue in V5 when serializing EventData to Variant types. 
Here, https://github.com/adobe/aepsdk-edge-ios/blob/dev/Sources/EdgeNetworkHandlers/RequestBuilder.swift#L111 If the EventData doesn't conform to AnyCodable, we drop the event.